spec/season_statistics_spec.rb

```diff
@@ -23,19 +23,32 @@
     end
   end
 
-#   describe "#winningest_coach" do
-#     xit " names the Coach with the best win percentage for the season" do 
-
-#       # return value string
-#     end
-#   end
-
-#   describe "#worst_coach" do
-#     xit " names the Coach with the worst win percentage for the season" do 
+  describe "#coach_record" do 
+  it "can list the loss, ties, and wins of each coach" do 
+    expected = {"Adam Oates" => {:games_coached=>183, :losses=>79, :ties=>44, :wins=>60},
+                "Alain Vigneault" => {:games_coached=>523, :losses=>192, :ties=>100, :wins=>231},
+                "Barry Trotz" => {:games_coached=>521, :losses=>179, :ties=>99, :wins=>243},
+                "Bill Peters" => {:games_coached=>328, :losses=>141, :ties=>62, :wins=>125},
+                "Bob Boughner" => {:games_coached=>82, :losses=>28, :ties=>16, :wins=>38},
+                "Bob Hartley" => {:games_coached=>305, :losses=>148, :ties=>52, :wins=>105},
+                "Bruce Boudreau" => {:games_coached=>511, :losses=>161, :ties=>108, :wins=>242}}
+    # expect(@season_stats.coach_record).to eq(expected)
```

Comment:
it looks like this "expected" variable is not holding all coaches that are printing in the method. When I stuck a pry in on ln 36 of season_stats and printed "coaches" it printed a super long list of coaches and their data. So I think the test is working for sure, but the test here is just not including everyone. 

I'd suggest making an assertion that the method results output a hash, and that a specific key is equal to a specific value. For example:

expect(@season_stats.coach_record).to be_a(Hash)

expect(@season_sats.coach_record["John Tortorella"].to eq({:games_coached=>392,

Reply:
You are absolutely correct, was having a brainfart, I have edited this test.